File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int small = prices[0];
        int max = 0;
        for(int i=1;i<prices.length;i++)
        {
            if(prices[i]<small)
                small=prices[i];
            max = Math.max(max,prices[i]-small);
        }
        return max;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution
{
    public:
        int maxProfit(vector<int> &prices)
        {
            int mini = 1e9;
            int maxans = 0;
            for (int i = 0; i < prices.size(); i++)
            {
                mini = min(prices[i],mini);
                maxans = max(maxans,prices[i]-mini);
            }
            return maxans;
        }
};